chaoscenter/web/src/views/ChaosInfrastructureReferenceField/ChaosInfrastructureReferenceField.tsx

```diff
@@ -46,32 +52,76 @@ interface ChaosInfrastructureReferenceFieldViewProps {
 
 function ChaosInfrastructureReferenceFieldView({
   infrastructureList,
+  environmentList,
+  allInfrastructureLength,
   preSelectedInfrastructure,
   setInfrastructureValue,
   searchInfrastructure,
   setSearchInfrastructure,
+  setEnvID,
+  activeEnv,
+  setActiveEnv,
   loading,
   pagination
 }: ChaosInfrastructureReferenceFieldViewProps): JSX.Element {
   const [isOpen, setOpen] = React.useState(false);
   const paths = useRouteWithBaseUrl();
   const history = useHistory();
+
   const [selectedInfrastructure, setSelectedInfrastructure] = React.useState<InfrastructureDetails | undefined>(
     preSelectedInfrastructure
   );
+
   // const searchParams = useSearchParams();
   // const infrastructureType =
   //   (searchParams.get('infrastructureType') as InfrastructureType | undefined) ?? InfrastructureType.KUBERNETES;
   const { showError } = useToaster();
   const { getString } = useStrings();
 
+  const listingEnvironment = ({ env }: { env: Environment }): JSX.Element => {
+    return (
+      <Container
+        key={env.environmentID}
+        flex
+        padding="small"
+        className={cx(css.listEnvContainer, activeEnv === env.environmentID && css.activeEnv)}
+        onClick={() => {
+          setEnvID(env.environmentID);
+          setActiveEnv(env.environmentID);
+        }}
+      >
+        <div className={css.itemEnv}>
+          <Layout.Horizontal padding={{ left: 'small' }} spacing="medium" className={css.leftInfo}>
+            <Text lineClamp={1} color={Color.GREY_800} font={{ variation: FontVariation.H6 }}>
+              {env.name}
+            </Text>
+          </Layout.Horizontal>
+          <Text
+            font={{ variation: FontVariation.SMALL }}
+            color={activeEnv === env.environmentID ? Color.WHITE : Color.PRIMARY_7}
```

Comment:
this condition is used at a lot of places, extract to a variable at the top of this component